engine/src/main/java/org/terasology/rendering/world/WorldRendererLwjgl.java

```diff
@@ -197,101 +266,97 @@ public void render(DefaultRenderingProcess.StereoRenderState stereoRenderState)
         switch (stereoRenderState) {
             case MONO:
                 currentRenderingStage = WorldRenderingStage.DEFAULT;
+                isFirstRenderingStageForCurrentFrame = true;
                 break;
             case OCULUS_LEFT_EYE:
                 currentRenderingStage = WorldRenderingStage.OCULUS_LEFT_EYE;
                 playerCamera.updateFrustum();
+                isFirstRenderingStageForCurrentFrame = true;
                 break;
             case OCULUS_RIGHT_EYE:
                 currentRenderingStage = WorldRenderingStage.OCULUS_RIGHT_EYE;
                 playerCamera.updateFrustum();
+                isFirstRenderingStageForCurrentFrame = false;
                 break;
         }
 
-        if (stereoRenderState == DefaultRenderingProcess.StereoRenderState.MONO || stereoRenderState == DefaultRenderingProcess.StereoRenderState.OCULUS_LEFT_EYE) {
-            renderableWorld.updateAndQueueVisibleChunks();
-        } else {
-            renderableWorld.updateAndQueueVisibleChunks(false, false);
-        }
-
-        if (renderingConfig.isDynamicShadows()
-                && (stereoRenderState == DefaultRenderingProcess.StereoRenderState.MONO || stereoRenderState == DefaultRenderingProcess.StereoRenderState.OCULUS_LEFT_EYE)) {
-            DefaultRenderingProcess.getInstance().beginRenderSceneShadowMap();
-            renderShadowMap();
-            DefaultRenderingProcess.getInstance().endRenderSceneShadowMap();
-        }
+        renderableWorld.updateAndQueueVisibleChunks(isFirstRenderingStageForCurrentFrame);
 
-        DefaultRenderingProcess.getInstance().beginRenderReflectedScene();
-        glCullFace(GL11.GL_FRONT);
-        playerCamera.setReflected(true);
+        renderShadowMap();
         renderWorldReflection();
-        playerCamera.setReflected(false);
-        glCullFace(GL11.GL_BACK);
-        DefaultRenderingProcess.getInstance().endRenderReflectedScene();
 
-        renderWorld();
+        toggleWireFrameON(); // if wireframe is enabled
```

Comment:
I would be tempted to put the code for enabling wire frame mode, along with the next two lines into some flavor of prepareForRendering() method. Methods that check and then do something are a bit awkward. Otherwise something like setWireFrameState() or checkAndActivateWireFrameMode() may be clearer.

Otherwise 'toggleWireFrameON()' should be 'toggleWireFrameOn()' since ON is not an initialisation/acronym. Likewise for OFF. But both names are bad since they do not reflect what the method does.